### PR TITLE
OPENE-368 Upload temporary docs while creating case.

### PR DESCRIPTION
**Pull request by jrudinskij**
Hi Arnas,

Could you please review when you get time?
Thanks.

### CODE REVIEW

**Comment by jrudinskij** [comment]
retest this please

File: openesdh-repo/src/main/java/dk/openesdh/repo/services/documents/CaseDocumentCopyServiceImpl.java [hunk 1]
```diff
@@ -220,6 +228,89 @@ public NodeRef copyDocFolderItemRetainVersionLabels(CaseFolderItem item, NodeRef
...
+
+    private NodeRef copyCaseDocFromTempAttachment(JSONObject tempAttachment, NodeRef targetFolderRef)
+            throws JSONException {
+        String tmpFileName = tempAttachment.getString(OeFilesService.TMP_FILE_NAME);
+        String fileName = tempAttachment.getString(OeFilesService.FILE_NAME);
+        String mimeType = tempAttachment.getString(OeFilesService.MIME_TYPE);
+        NodeRef docType = new NodeRef(tempAttachment.getString(WebScriptParams.DOC_TYPE));
+        NodeRef docCategory = new NodeRef(tempAttachment.getString(WebScriptParams.DOC_CATEGORY));
+
+        File[] files = TempFileProvider.getTempDir().listFiles((file, name) -> name.equals(tmpFileName));
```
Comment (ArnasPetraitis) [inline openesdh-repo/src/main/java/dk/openesdh/repo/services/documents/CaseDocumentCopyServiceImpl.java]: this few lines could be extracted to getFileByName

File: openesdh-repo/src/main/java/dk/openesdh/repo/services/documents/CaseDocumentCopyServiceImpl.java [hunk 2]
```diff
@@ -220,6 +228,89 @@ public NodeRef copyDocFolderItemRetainVersionLabels(CaseFolderItem item, NodeRef
...
+        if (files.length == 0) {
+            throw new AlfrescoRuntimeException("Temp attachment file not found " + tmpFileName + " " + fileName);
+        }
+
+        Map<QName, Serializable> props = new HashMap<>();
```
Comment (ArnasPetraitis) [inline openesdh-repo/src/main/java/dk/openesdh/repo/services/documents/CaseDocumentCopyServiceImpl.java]: all the following could be replaced to call to documentService.createDocumentFile

File: openesdh-repo/src/main/java/dk/openesdh/repo/webscripts/files/OeFilesWebScript.java [hunk 3]
```diff
@@ -115,6 +123,32 @@ public void addFileToCase(
         filesService.addToCase(caseId, file, title, docType, docCategory, description);
     }
 
+    @Uri(value = "/api/openesdh/files/upload/tmp", multipartProcessing = true, method = HttpMethod.POST, defaultFormat = WebScriptUtils.JSON)
```
Comment (ArnasPetraitis) [inline openesdh-repo/src/main/java/dk/openesdh/repo/webscripts/files/OeFilesWebScript.java]: My intension of "OeFilesWebScript" was for user's files only. I sugest to move those actions to own TempFilesWebScript or someting similar

**Pull request merged**